address_restricted_ed25519.go

```diff
@@ -0,0 +1,241 @@
+package iotago
+
+import (
+	"bytes"
+	"crypto/ed25519"
+
+	"golang.org/x/crypto/blake2b"
+
+	"github.com/iotaledger/hive.go/ierrors"
+	"github.com/iotaledger/hive.go/serializer/v2"
+	"github.com/iotaledger/hive.go/serializer/v2/byteutils"
+	"github.com/iotaledger/iota.go/v4/hexutil"
+)
+
+const (
+	// RestrictedEd25519AddressMinBytesLength is the min length of a restricted Ed25519 address.
+	RestrictedEd25519AddressMinBytesLength        = Ed25519AddressBytesLength + serializer.OneByte
+	RestrictedEd25519AddressMaxCapabilitiesLength = 1
+)
+
+const (
+	canReceiveNativeTokensBitIndex = iota
+	canReceiveManaBitIndex
+	canReceiveOutputsWithTimelockUnlockConditionBitIndex
+	canReceiveOutputsWithExpirationUnlockConditionBitIndex
+	canReceiveOutputsWithStorageDepositReturnUnlockConditionBitIndex
+	canReceiveAccountOutputsBitIndex
+	canReceiveNFTOutputsBitIndex
+	canReceiveDelegationOutputsBitIndex
+)
+
+type RestrictedEd25519Address struct {
+	PubKeyHash   [Ed25519AddressBytesLength]byte `serix:"0"`
+	Capabilities []byte                          `serix:"1,lengthPrefixType=uint8,maxLen=1"`
```

Comment:
```suggestion
	AllowedCapabilities []byte                          `serix:"1,lengthPrefixType=uint8,maxLen=1"`
```

I think calling it `AllowedCapabilites` would be clearer to indicate that it's an AllowList rather than a DenyList.